coding_systems/icd10/coding_system.py

```diff
@@ -20,18 +20,17 @@ def search_by_term(term):
 
 
 def search_by_code(code):
-    try:
-        concept = Concept.objects.get(code__iexact=code)
-    except Concept.ObjectDoesNotExist:
-        return set()
-
-    # The UI does not support top-level concepts being included/excluded (and in any
-    # case, they cannot appear on a patient record) so if a user searches for a chapter
-    # code, we instead return the codes for all children of that chapter.
-    if concept.kind == "chapter":
-        return {child.code for child in concept.children.all()}
+    code = code.upper()
+    if code[-1] == "*":
+        kwargs = {"code__startswith": code[:-1]}
```

Comment:
nano-nit: If you used `code.endswith("*")` you'd be robust against having an empty string (although presumably that's enforced elsewhere. And you used `code.rstrip("*")` you'd be robust against a user accidentally supplying multiple wildcards.

Reply:
Nice idea.